commands/scan/scan.go

```diff
@@ -246,9 +263,22 @@ func (scanCmd *ScanCommand) Run() (err error) {
 	scanErrors = appendErrorSlice(scanErrors, fileProducerErrors)
 	scanErrors = appendErrorSlice(scanErrors, indexedFileProducerErrors)
 
-	scanResults := xrutils.NewAuditResults()
-	scanResults.XrayVersion = xrayVersion
-	scanResults.ScaResults = []xrutils.ScaScanResult{{XrayResults: flatResults}}
+	// Wait for the Download of the AnalyzerManager to complete.
+	if err = errGroup.Wait(); err != nil {
+		err = errors.New("failed while trying to get Analyzer Manager: " + err.Error())
+	}
+
+	if scanResults.ExtendedScanResults.EntitledForJas {
+		depsList := depsListFromVulnerabilities(flatResults)
+
+		for _, scanResult := range flatResults {
+			scanResults.ScaResults = append(scanResults.ScaResults, xrutils.ScaScanResult{XrayResults: []services.ScanResponse{scanResult}, Technology: coreutils.Technology(scanResult.ScannedPackageType)})
+		}
+
+		workingDirs := []string{scanCmd.spec.Files[0].Pattern}
```

Comment:
The `scan` command can get multiple binaries to scan.
It could get a pattern of files as well.
In those cases this will not work.
We are walking and collecting the files and running the SCA scan on each of them separated.
The JAS scan should also run for each binary.

Reply:
The scan command is not in the scope of this PR only docker-scan